### ISSUE TITLE
wrongUsernamePassword not returned for invalid logins for password grant flow

### ISSUE THREAD

**Issue by pivotal-ryan-mcleod** [p2/OAuth2#223]
It seems that the OAuth2PasswordGrant expects a 403 (or 401) status code to be returned when the user provides invalid credentials to the OAuth2 service, but in fact the OAuth 2 specification states that it will actually return a 400 with an error of "invalid_grant". https://tools.ietf.org/html/rfc6749#section-5.2

As a result, when I am sending bad credentials to my OAuth server (using Spring's default OAuth implementation http://projects.spring.io/spring-security-oauth/) I'm getting a `responseError` with a description of "Bad Credentials" instead of a `wrongUsernamePassword` error.

**Comment by p2** [comment]
Yes, should detect `invalid_grant` in `error` and raise the same error!
